### PR TITLE
Add management workload annotations

### PR DESCRIPTION
**Pull request by lack**
In support of the workload partitioning feature
(https://github.com/openshift/enhancements/pull/703), we need to add
annotations to all management pods and namespaces so they can be
properly identified and assigned to segregated management cores on
clusters configured to do so.

### CODE REVIEW

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

**Comment by ricardomaraschini** [comment]
/hold

Let's figure out if the tests are passing then we unhold it.

**Comment by lack** [comment]
Rebased to catch the addition of `manifests/07-operator-ibm-cloud-managed.yaml`

**Comment by lack** [comment]
FYI: The associated bugzilla is https://bugzilla.redhat.com/show_bug.cgi?id=1949598

**Comment by lack** [comment]
/retest
:face_with_head_bandage:

**Comment by lack** [comment]
@ricardomaraschini : Do you have any insight into this recurring failure?  I really don't think adding these annotations (that no one looks at unless you have a very special machineconfig added to a cluster) should have any effect at all.

/retest

**Comment by imiller0** [comment]
/unhold
openshift/kubernetes#632 has now merged, please review and merge this PR as well.

**Comment by lack** [comment]
@ricardomaraschini This is all ready to go and just needs a new LGTM from you please :)

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

**Pull request merged**
